src/app/shared/services/vm-snapshot.service.ts

```diff
@@ -43,10 +56,9 @@ export class VmSnapshotService extends BaseBackendService<VmSnapshot> {
       );
   }
 
-  public revert(id: string): Observable<any> {
-    // todo it returns vm, maybe we need to update vms
+  public revert(id: string): Observable<VirtualMachine> {
     return super
       .sendCommand('revertTo', { vmsnapshotid: id })
-      .pipe(switchMap(job => this.asyncJobService.queryJob(job, this.entity)));
+      .pipe(switchMap(job => this.asyncJobService.queryJob(job, virtualMachineEntityName)));
```

Comment:
It's strange to see `virtualMachineEntityName` here since it's related to the other service, which leads to broken encapsulation principles.